src/utils/user.authentication.ts

```diff
@@ -10,11 +10,20 @@ import {promisify} from 'util';
 import * as isemail from 'isemail';
 import {HttpErrors} from '@loopback/rest';
 import {UserProfile} from '@loopback/authentication';
-import {compare} from 'bcryptjs';
-const compareAsync = promisify(compare);
+import {genSalt, compare, hash} from 'bcryptjs';
 const jwt = require('jsonwebtoken');
 const signAsync = promisify(jwt.sign);
 const verifyAsync = promisify(jwt.verify);
+const hashAsync = promisify(hash);
+const compareAsync = promisify(compare);
```

Comment:
I believe both `hash` and `compare` return promises when no callback was provided. Could you please try to `await` them directly without `promisify` wrapper?

API docs:

https://github.com/dcodeIO/bcrypt.js#hashs-salt-callback-progresscallback
> `@returns | Promise | If callback has been omitted`

https://github.com/dcodeIO/bcrypt.js#compares-hash-callback-progresscallback
> `@returns | Promise | If callback has been omitted`